### PR TITLE
Configure SA1201 action as error

### PR DESCRIPTION
**Pull request by BL323**
The intention of this change is to enable the analyzer rule [SA1201: ElementsMustAppearInTheCorrectOrder](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1201.md) as described in the issue #1388.

Please let me know if there is anything I've missed while raising the PR.

### CODE REVIEW

File: lib/PuppeteerSharp/Browser.cs [hunk 1]
```diff
@@ -42,6 +42,15 @@ public class Browser : IDisposable, IAsyncDisposable
         /// </summary>
         private const int CloseTimeout = 5000;
 
+        #region Private members
+
```
Comment (kblok) [inline lib/PuppeteerSharp/Browser.cs:46]: Can you remove the regions?
I'm taking the chance to them every time I see one :)

**Review by kblok — APPROVED** [review]
Thank you!

File: lib/PuppeteerSharp/Browser.cs (hunk 1 above)
Comment (BL323) [reply lib/PuppeteerSharp/Browser.cs]: All done :) I've now removed them in few files that were in the vicinity of my previous changes.

I could look at picking up the issue #1395 to enable SA1124 which would remove all the other regions. Although I didn't want to make the scope of this PR too big.

**Comment by BL323** [comment]
The failing test `NetworkRedirectsShouldReportSecurityDetails` seems to be a little flaky, Sometimes this passes, sometimes fails when running from my local machine. 

I don't think any of the from this PR would have effected it's behavior?

**Pull request merged**